apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl

```diff
@@ -249,23 +244,52 @@ t_downsample_1h(_Config) ->
     MaxAge = timer:hours(1),
     test_downsample(MaxAge, 10).
 
-sent_1() -> #{sent => 1}.
+%% Since the monitor process is running, and tests like `t_downsample_*' expect some
+%% degree of determinism, we need to pause that process to avoid having it insert a rogue
+%% point amidst the "carefully crafted" dataset.
+pause_monitor_process() ->
+    ok = sys:suspend(emqx_dashboard_monitor),
+    on_exit(fun() -> ok = sys:resume(emqx_dashboard_monitor) end),
+    ok.
+
+sent_n(N) -> #{sent => N}.
+sent_1() -> sent_n(1).
+
+%% a gauge
+connections_n(N) -> #{connections => N}.
+connections_1() -> connections_n(1).
 
 round_down(Ts, Interval) ->
     Ts - (Ts rem Interval).
 
 test_downsample(MaxAge, DataPoints) ->
+    ok = pause_monitor_process(),
     Now = erlang:system_time(millisecond) - 1,
     Interval = emqx_dashboard_monitor:sample_interval(MaxAge),
     StartTs = round_down(Now - MaxAge, Interval),
-    ok = emqx_dashboard_monitor:clean(0),
+    clean_data(),
     %% insert the start mark for deterministic test boundary
-    ok = write(StartTs, sent_1()),
-    ok = insert_data_points(DataPoints - 1, StartTs, Now),
```

Comment:
This was a source of flakiness: `Now` could actually change the `sample_interval` together with the computed `StartTs` from the expected `Interval`, making "strange" timestamps appear.